src/nvim/eval/funcs.c

```diff
@@ -5404,8 +5404,17 @@ static void f_jobwait(typval_T *argvars, typval_T *rettv, FunPtr fptr)
   TV_LIST_ITER_CONST(args, arg, {
     Channel *chan = NULL;
     if (TV_LIST_ITEM_TV(arg)->v_type != VAR_NUMBER
-        || !(chan = find_job(TV_LIST_ITEM_TV(arg)->vval.v_number, false))) {
+        || !(chan = find_channel(TV_LIST_ITEM_TV(arg)->vval.v_number))
+        || chan->streamtype != kChannelStreamProc) {
       jobs[i] = NULL;  // Invalid job.
+    } else if (process_is_stopped(&chan->stream.proc)) {
+      // Process is already stopped, so ensure all callbacks on its event
+      // queue are drained
+      Process *proc = &chan->stream.proc;
+      LOOP_PROCESS_EVENTS_UNTIL(proc->loop, proc->events, -1, !proc->refcount);
+
+      // Set job to NULL so that the return value will be set to -3 below
+      jobs[i] = NULL;
```

Comment:
as a reader it would help me see a pattern if this stays identical with the case above. the longer comment doesn't seem to add anything.
```suggestion
      jobs[i] = NULL;  // Invalid job.
```

Reply:
I avoided that language because I didn’t think a stopped job was “invalid”, but I also don’t feel strongly about it and I certainly want to improve readability if possible. So this is fine with me.